sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/AspNetHostedService/src/EventProcessorClientService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace Azure.Messaging.EventHubs.Processor.HostedService.Samples
+{
+    /// <summary>
+    /// Implementation of the <see cref="IHostedService"/> interface for processing events.
+    /// </summary>
+    public class EventProcessorClientService : IHostedService
+    {
+        private readonly ILogger<EventProcessorClientService> _logger;
+        private readonly EventProcessorClient _processorClient;
+        private readonly ISampleApplicationProcessor _appProcessor;
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="EventProcessorClientService"/> class.
+        /// </summary>
+        /// <param name="logger">A named <see cref="ILogger"/> used for logging within the <see cref="EventProcessorClientService"/> class.</param>
+        /// <param name="processorClient"><see cref="EventProcessorClient"/> used for consuming and processing events within the <see cref="EventProcessorClientService"/> class.</param>
+        /// <param name="appProcessor">A named <see cref="ISampleApplicationProcessor"/> used as an example to show how to pass events to a downstream service for further processing.</param>
+        public EventProcessorClientService(
+            ILogger<EventProcessorClientService> logger,
+            EventProcessorClient processorClient,
+            ISampleApplicationProcessor appProcessor)
+        {
+            _logger = logger;
+            _processorClient = processorClient;
+            _appProcessor = appProcessor;
+        }
+
+        /// <summary>
+        /// Implementation of the <see cref="IHostedService"/> StartAsync interface method.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Initialize the client event handlers.
+                _processorClient.PartitionInitializingAsync += InitializeEventHandler;
+                _processorClient.ProcessEventAsync += ProcessEventHandler;
+                _processorClient.ProcessErrorAsync += ProcessErrorHandler;
+
+                //Start processing events.
+                await _processorClient.StartProcessingAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch
+            {
+                // It is important that you guard against exceptions in
```

Comment:
This is true of the event handlers assigned to the processor, but I think we want to throw here if the processor has issues starting, no?   For example, if the connection string is incorrect, we know that the processor will never recover.